libs/community/langchain_community/retrievers/zep_cloud.py

```diff
@@ -0,0 +1,165 @@
+from __future__ import annotations
+
+from typing import Any, Dict, List, Optional
+
+from langchain_core.callbacks import (
+    AsyncCallbackManagerForRetrieverRun,
+    CallbackManagerForRetrieverRun,
+)
+from langchain_core.documents import Document
+from langchain_core.pydantic_v1 import root_validator
+from langchain_core.retrievers import BaseRetriever
+
+try:
+    from zep_cloud import MemorySearchResult, SearchScope, SearchType
+    from zep_cloud.client import AsyncZep, Zep
+
+    class ZepCloudRetriever(BaseRetriever):
+        """`Zep Cloud` MemoryStore Retriever.
```

Comment:
this type of definition will prevent this class from being properly documented in the [API reference](https://api.python.langchain.com/en/latest/community_api_reference.html). Would it be better to just remove the typing from the pydantic attributes (eg `zep_client: Any` instead of `zep_client: Zep`)?

Reply:
Roger that, addressed. Also did the same for async client.

Reply:
sorry should've been clearer, we need to remove the class definition from the try/except clause for it to be documented

Reply:
@baskaryan my bad, removed the try/except clause.